test/e2e/aks_adopt.go

```diff
@@ -0,0 +1,154 @@
+//go:build e2e
+// +build e2e
+
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+
+	. "github.com/onsi/gomega"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	clusterctlv1 "sigs.k8s.io/cluster-api/cmd/clusterctl/api/v1alpha3"
+	expv1 "sigs.k8s.io/cluster-api/exp/api/v1beta1"
+	"sigs.k8s.io/cluster-api/test/framework/clusterctl"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type AKSAdoptSpecInput struct {
+	ApplyInput   clusterctl.ApplyClusterTemplateAndWaitInput
+	ApplyResult  *clusterctl.ApplyClusterTemplateAndWaitResult
+	Cluster      *clusterv1.Cluster
+	MachinePools []*expv1.MachinePool
+}
+
+// AKSAdoptSpec tests adopting an existing AKS cluster into management by CAPZ. It first relies on a CAPZ AKS
+// cluster having already been created. Then, it will orphan that cluster such that the CAPI and CAPZ
+// resources are deleted but the Azure resources remain. Finally, it applies the cluster template again and
+// waits for the cluster to become ready.
+func AKSAdoptSpec(ctx context.Context, inputGetter func() AKSAdoptSpecInput) {
+	input := inputGetter()
+
+	mgmtClient := bootstrapClusterProxy.GetClient()
+	Expect(mgmtClient).NotTo(BeNil())
+
+	updateResource := []any{"30s", "5s"}
+
+	waitForNoBlockMove := func(obj client.Object) {
+		waitForBlockMoveGone := []any{"30s", "5s"}
```

Comment:
Should these intervals be moved to the intervals section in azure-dev? https://github.com/willie-yao/cluster-api-provider-azure/blob/main/test/e2e/config/azure-dev.yaml#L231

Reply:
These values are pretty specific to this test only, so to me it seems like it's fine to keep them local to this file. Otherwise I always feel like I have a hard time tracking down what these values are.